src/test/java/org/prebid/server/auction/PrivacyEnforcementServiceTest.java

```diff
@@ -0,0 +1,615 @@
+
+    @Mock
+    private BidderCatalog bidderCatalog;
+    @Mock
+    private Metrics metrics;
+    @Mock
+    private GdprService gdprService;
+
+    private Clock clock;
+    private Timeout timeout;
+    private PrivacyEnforcementService privacyEnforcementService;
+
+    @Before
+    public void setUp() {
+        given(bidderCatalog.bidderInfoByName(anyString())).willReturn(givenBidderInfo(15, true));
+
+        clock = Clock.fixed(Instant.now(), ZoneId.systemDefault());
+        timeout = new TimeoutFactory(clock).create(500);
+
+        privacyEnforcementService = new PrivacyEnforcementService(gdprService, bidderCatalog, metrics, false);
+    }
+
+    @Test
+    public void shouldNotMaskWhenDeviceLmtIsNullAndExtRegsGdprIsOneAndGdprServiceRespondIsGdprEnforcedWithTrueAndResultByVendorWithoutEnforcement() {
+        // given
+        given(gdprService.isGdprEnforced(eq("1"), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout)))
+                .willReturn(Future.succeededFuture(true));
+        given(gdprService.resultByVendor(eq(singleton(15)), eq("1"), any(), any(), eq(timeout)))
+                .willReturn(Future.succeededFuture(GdprResponse.of(true, singletonMap(15, true), null)));
+
+        final ExtUser extUser = ExtUser.builder().build();
+        final User user = notMaskedUser();
+        final Device device = notMaskedDevice();
+        final ExtRegs extRegs = ExtRegs.of(1);
+        final Regs regs = Regs.of(null, mapper.valueToTree(extRegs));
+        final Map<String, User> bidderToUser = singletonMap(BIDDER_NAME, user);
+
+        final BidRequest bidRequest = givenBidRequest(givenSingleImp(
+                singletonMap(BIDDER_NAME, 1)),
+                bidRequestBuilder -> bidRequestBuilder
+                        .user(user)
+                        .device(device)
+                        .regs(regs));
+
+        // when
+        final Future<Map<String, PrivacyEnforcementResult>> masked = privacyEnforcementService
+                .mask(bidderToUser, extUser, extRegs, singletonList(BIDDER_NAME), emptyMap(), bidRequest, PUBLISHER_ID, timeout);
+
+        // then
+        final PrivacyEnforcementResult expected = PrivacyEnforcementResult.of(user, device, regs);
+
+        verify(gdprService).isGdprEnforced(eq("1"), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout));
+        verify(gdprService).resultByVendor(eq(singleton(15)), eq("1"), any(), any(), eq(timeout));
+        verifyNoMoreInteractions(gdprService);
+
+        assertThat(masked.succeeded()).isTrue();
+        assertThat(masked.result()).hasSize(1)
+                .containsOnly(entry(BIDDER_NAME, expected));
+    }
+
+    @Test
+    public void shouldNotMaskWhenExtDeviceLmtIsNullAndGdprServiceRespondIsGdprEnforcedWithFalse() {
+        // given
+        given(gdprService.isGdprEnforced(isNull(), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout)))
+                .willReturn(Future.succeededFuture(false));
```

Comment:
Instead of declaring a stub for each method - declare a common stub (that accepts any parameters) for most used cases (like returning future with `false` value).

Declare another mock (future with `true`) where needed.

Also, stubbing like this makes further `verify` meaningless, since your stub is very specific and the only case test passes is when this stub is matched, that is - expected params were used.